src/components/Unicornscan/Unicornscan.tsx

```diff
@@ -0,0 +1,238 @@
+                setLoadingModal(false); // Also set loading to false in case of error
+            });
+    }, []);
+    /**
+     * handleProcessData: Callback to handle and append new data from the child process to the output.
+     * It updates the state by appending the new data received to the existing output.
+     * @param {string} data - The data received from the child process.
+     */
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data);
+    }, []);
+    /**
+     * handleProcessTermination: Callback to handle the termination of the child process.
+     * Once the process termination is handled, it clears the process PID reference and
+     * deactivates the loading overlay.
+     * @param {object} param - An object containing information about the process termination.
+     * @param {number} param.code - The exit code of the terminated process.
+     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     */
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+            setPid("");
+            setLoading(false);
+        },
+        [handleProcessData]
+    );
+    /**
+     * onSubmit: Asynchronous handler for the form submission event.
+     * It sets up and triggers the tshark tool with the given parameters.
+     * Once the command is executed, the results or errors are displayed in the output.
+     *
+     * @param {FormValuesType} values - The form values, containing the <list the form values here, e.g.  interface, packet count, etc>.
+     */
+    const onSubmit = async (values: FormValuesType) => {
+        setLoading(true);
+        let args = [values.targetIP];
+
+        // Add the selected scan type to the arguments
+        if (selectedScanType === "TCP Scan (-mT)") {
+            args.push("-mT");
+        } else if (selectedScanType === "UDP Scan (-mU)") {
+            args.push("-mU");
+        }
+
+        // Include additional flags if they are set
+        if (rate) args.push(`-r ${rate}`);
+        if (sourcePort) args.push(`-e ${sourcePort}`);
+        if (sourceIP) args.push(`-g ${sourceIP}`);
+        if (interfaceName) args.push(`-i ${interfaceName}`);
+        if (ports) args.push(`-p ${ports}`);
+        if (verboseMode) args.push("-v");
+
+        CommandHelper.runCommandWithPkexec("unicornscan", [...args, "-I"], handleProcessData, handleProcessTermination)
+            .then(() => {
+                setLoading(false);
```

Comment:
I found that the loadingOverlayAndCancel screen was not appearing, so copied Aircrack-NG, which also uses runCommandWithPkexec. I found that once I made these changes, the loading screen started to appear. 
```suggestion
               setOutput(output);
                setAllowSave(true);
                setPid(pid);
```